wodin_deploy/wodin_constellation.py

```diff
@@ -0,0 +1,117 @@
+from constellation import Constellation, ConstellationContainer, ConstellationMount, docker_util
+
+
+class WodinConstellation:
+    def __init__(self, cfg):
+        self.cfg = cfg
+        redis = redis_container(cfg)
+        odin_api = odin_api_container(cfg)
+        sites = sites_containers(cfg)
+        proxy = wodin_proxy_container(cfg)
+
+        containers = [redis, odin_api, *sites, proxy]
+
+        self.obj = Constellation(
+            "wodin",
+            cfg.container_prefix,
+            containers,
+            cfg.network,
+            {"redis-data": "redis-data", "wodin-config": "wodin-config"},
+            data=cfg,
+        )
+
+    def start(self, **kwargs):
+        self.obj.start(**kwargs)
+
+    def stop(self, **kwargs):
+        self.obj.stop(**kwargs)
+
+    def status(self):
+        self.obj.status()
+
+
+def redis_container(cfg):
+    redis = cfg.redis
+    redis_mounts = [ConstellationMount("redis-data", "/data")]
+    return ConstellationContainer(redis["name"], redis["ref"], mounts=redis_mounts, network=cfg.network)
+
+
+def odin_api_container(cfg):
+    odin_api = cfg.odin_api
+    return ConstellationContainer("api", odin_api["ref"], network=cfg.network)
+
+
+def get_wodin_container(cfg, site, site_dict):
+    wodin = cfg.wodin
+    prefix = cfg.container_prefix
+    wodin_mounts = [ConstellationMount("wodin-config", "/wodin/config")]
+    container_name = f"{wodin['name']}-{site}"
+
+    return ConstellationContainer(
+        container_name,
+        wodin["ref"],
+        mounts=wodin_mounts,
+        network=cfg.network,
+        entrypoint=[
+            "/wodin/docker/pull-site-and-start.sh",
```

Comment:
So this entrypoint will cause each wodin container to pull the config for its own site and will save that into the `wodin-config` volume which they all share - but each will have its own folder for its corresponding repo within the volume, so there shouldn't be any clashes.?

Reply:
yep exactly, each site will be separated in the volume by path "/{site}" so they will just be git cloning to different folders whenever the containers come up